diff --git a/ui/base/clipboard/clipboard_aurax11.cc b/ui/base/clipboard/clipboard_aurax11.cc
index 39e864738..181db2211 100644
--- a/ui/base/clipboard/clipboard_aurax11.cc
+++ b/ui/base/clipboard/clipboard_aurax11.cc
@@ -23,6 +23,7 @@
 #include "base/strings/utf_string_conversions.h"
 #include "third_party/skia/include/core/SkBitmap.h"
 #include "ui/base/clipboard/clipboard_monitor.h"
+#include "ui/base/clipboard/clipboard_qt_mapper.h"
 #include "ui/base/clipboard/custom_data_helper.h"
 #include "ui/base/x/selection_owner.h"
 #include "ui/base/x/selection_requestor.h"
@@ -859,8 +860,10 @@ void ClipboardAuraX11::WriteObjects(ClipboardType type,
       const ObjectMapParams& params_vector = text_iter->second;
       if (params_vector.size()) {
         const ObjectMapParam& char_vector = params_vector[0];
-        if (char_vector.size())
+        if (char_vector.size()) {
           WriteText(&char_vector.front(), char_vector.size());
+          WriteTextToQt(&char_vector.front(), char_vector.size());
+        }
       }
       aurax11_details_->TakeOwnershipOfSelection(CLIPBOARD_TYPE_SELECTION);
     }
diff --git a/ui/base/clipboard/clipboard_qt_mapper.cc b/ui/base/clipboard/clipboard_qt_mapper.cc
new file mode 100644
index 000000000..f1e9dd202
--- /dev/null
+++ b/ui/base/clipboard/clipboard_qt_mapper.cc
@@ -0,0 +1,26 @@
+
+#include "ui/base/clipboard/clipboard_qt_mapper.h"
+
+#include <stdio.h>
+
+namespace ui {
+
+namespace {
+
+ClipboardQtDelegate* g_delegate = nullptr;
+
+}  // namespace
+
+void WriteTextToQt(const char* text_data, size_t text_len) {
+  if (g_delegate != nullptr) {
+    g_delegate->OnTextChanged(text_data, text_len);
+  }
+}
+
+void InstallClipboardQtDelegate(ClipboardQtDelegate* delegate) {
+  if (delegate != nullptr) {
+    g_delegate = delegate;
+  }
+}
+
+}  // namespace ui
diff --git a/ui/base/clipboard/clipboard_qt_mapper.h b/ui/base/clipboard/clipboard_qt_mapper.h
new file mode 100644
index 000000000..1d369bee0
--- /dev/null
+++ b/ui/base/clipboard/clipboard_qt_mapper.h
@@ -0,0 +1,23 @@
+
+#ifndef UI_BASE_CLIPBOARD_CLIPBOARD_QT_MAPPER_H_
+#define UI_BASE_CLIPBOARD_CLIPBOARD_QT_MAPPER_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "ui/base/ui_base_export.h"
+
+namespace ui {
+
+UI_BASE_EXPORT void WriteTextToQt(const char* text_data, size_t text_len);
+
+class UI_BASE_EXPORT ClipboardQtDelegate {
+ public:
+  virtual void OnTextChanged(const char* text_data, size_t text_len) = 0;
+};
+
+UI_BASE_EXPORT void InstallClipboardQtDelegate(ClipboardQtDelegate* delegate);
+
+}  // namespace ui
+
+#endif  // UI_BASE_CLIPBOARD_CLIPBOARD_QT_MAPPER_H_
